modules/common/hardware/lenovo-x1/definitions/default.nix

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+{
+  generation,
+  lib,
+}: let
+  hwDefinition = import (./. + "/x1-${generation}.nix");
+in {
+  inherit (hwDefinition) mouse;
+  inherit (hwDefinition) touchpad;
+  inherit (hwDefinition) disks;
+  inherit (hwDefinition) network;
+  inherit (hwDefinition) gpu;
+
+  # Notes:
+  #   1. This assembles udev rules for different hw configurations (i.e., different mice/touchpads) by adding
+  #      all of them to the configuration. This was chosen for simplicity to not have to provide hw identifier at build,
+  #      but is not ideal and should be changed.
+  #   2. USB camera "passthrough" is handled by qemu and thus available on host. If peripheral VM is implemented,
+  #      the entire host controller should be passthrough'd using the PCI bus (14.0). In x1, bluetooth and fingerprint
+  #      reader are on this bus.
+  udevRules = let
+    mapMouseRules =
+      builtins.map (d: ''        SUBSYSTEM=="input", ATTRS{name}=="${d}", KERNEL=="event*", GROUP="kvm", SYMLINK+="mouse"
+      '');
+    mapTouchpadRules =
+      builtins.map (d: ''        SUBSYSTEM=="input", ATTRS{name}=="${d}", KERNEL=="event*", GROUP="kvm", SYMLINK+="touchpad"
+      '');
+  in ''
+    # Laptop keyboard
+    SUBSYSTEM=="input", ATTRS{name}=="AT Translated Set 2 keyboard", GROUP="kvm"
+    # Laptop TrackPoint
+    SUBSYSTEM=="input", ATTRS{name}=="TPPS/2 Elan TrackPoint", GROUP="kvm"
+    # Lenovo X1 integrated webcam
+    KERNEL=="3-8", SUBSYSTEM=="usb", ATTR{busnum}=="3", ATTR{devnum}=="3", GROUP="kvm"
+    # Mouse and Touchpad
+    ${lib.strings.concatStrings (mapMouseRules hwDefinition.mouse)}
```

Comment:
We potentially also want to include Human interface devices in generally here?

```
Bus 001 Device 043: ID 046d:c548 Logitech, Inc. Logi Bolt Receiver
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 [unknown]
  bDeviceSubClass         0 [unknown]
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x046d Logitech, Inc.
  idProduct          0xc548 Logi Bolt Receiver
  bcdDevice            5.01
  iManufacturer           1 Logitech
  iProduct                2 USB Receiver
  iSerial                 0
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          4
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower               98mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device <-----
```

Reply:
There is ID_TYPE=hid, but not sure if it's too broad. It would also include my jabra headphone dongle.

Reply:
`ID_INPUT_KEYBOARD=1` and `ID_INPUT_MOUSE=1` might work.

Reply:
But instead of adding the KVM group here, could the VM that is responsible for input not be in the input group? This one already exists.

Reply:
I think at the moment all the VMs have the same user microvm and require the input, hence the adding to KVM group for microvm user, but adding microvm to input may also be an option. ID_INPUT_* may actually a good solution here - haven't really looked into that here. 
If we move to a dedicated peripheral VM, the host passthroughs+udev rules become trivial for usb devices if we simply pass the controllers.